### PR TITLE
Issue #1468582 by Schnitzel, lewisnyman, dcmouyard, alanburke, tstoeckler, edward_or, janusman: Add mobile friendly meta tags to the html.tpl.php.

### PR DESCRIPTION
**Pull request by hellomrcat**
PR for https://github.com/backdrop/backdrop-issues/issues/200

### CODE REVIEW

**Comment by quicksketch** [comment]
Nice! I'll need to have @jenlampton review this one. It's great to get these in here but we'll have to re-evaluate the approach. D8 eventually went with the equivalent of backdrop_add_html_head() to add these tags, meaning they all ended up as part of the single $head variable.

File: core/modules/system/tests/system_module_test.module [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * @file
+ * Provides System module hook implementations for testing purposes.
+ */
+
+function system_module_test_preprocess_html(&$variables) {
+  $variables['default_mobile_metatags'] = FALSE;
+}
+?>
```
Comment (jenlampton) [inline core/modules/system/tests/system_module_test.module]: We should remove this closing PHP tag.

File: core/modules/system/tests/system_module_test.info [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name = "System test"
+description = "Provides hook implementations for testing System module functionality."
+package = Testing
+version = BACKDROP_VERSION
+backdrop = 1.x
+hidden = TRUE
```
Comment (jenlampton) [inline core/modules/system/tests/system_module_test.info]: We should add an end-of-line marker here.

**Comment by jenlampton** [comment]
Yeah, I don't like the extra variable added into the template.  If we removed that the changes in core/includes/theme.inc and core/modules/system/html.tpl.php would go away.

We'll need to add a change in common.inc, to backdrop_get_html_head, as follows.

```
/**
 * Retrieves output to be displayed in the HEAD tag of the HTML page.
 */
function backdrop_get_html_head() {
  $elements = backdrop_add_html_head();

  // Always add the default mobile meta tags.
  $elements['common-viewport'] = array(
    '#type' => 'html_tag',
    '#tag' => 'meta',
    '#attributes' => array(
      'name' => 'viewport',
      'content' => 'width=device-width',
    ),
  );
  $elements['common-cleartype'] = array(
    '#type' => 'html_tag',
    '#tag' => 'meta',
    '#attributes' => array(
      'http-equiv' => 'cleartype',
      'content' => 'on',
    ),
  );

  backdrop_alter('html_head', $elements);
  return backdrop_render($elements);
}
```

@hellomrcat  can you take a stab at this one, and see if the same test from this issue still passes?

**Comment by quicksketch** [comment]
Super! It looks like D8 ultimately removed the cleartype meta tag in https://www.drupal.org/node/1717090, so I dropped that in the merge. Previously we had Seven adding the mobile meta tag, but that's not necessary if core is adding it. I removed that also as part of the merge. Thanks!

**Pull request merged**